### PR TITLE
refactor(didit-sdk): wagmi is not required if wallet auth is not used

### PR DESCRIPTION
**Pull request by Keddib**
changes: 
- [x] wagmi is not required when not using wallet auth method
- [x] update documentation

BREAKING CHANGE: you need to remove DiditRainbowKitProvider and pass all props {theme, chains, ... } to DiditAuthProvider

Now you can use the SDK without using wagmi if you don't want to use the wallet method

and DiditRainbowkitProvider is declared internally : 

Screenshot 2023-10-31 at 16 24 08

### CODE REVIEW

File: packages/didit-sdk/src/hooks/useDiditAuth.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ const useDiditAuth = ({
     useDiditEmailAuthContext();
   const { adapter, address } =
     useContext(RainbowKitAuthenticationContext) ?? {};
+  // TODO: useConnectModal will throw an error if we are noting using wallet auth method
```
Comment (pablogamium) [inline packages/didit-sdk/src/hooks/useDiditAuth.ts:34]: which makes sense

**Review by pablogamium — COMMENTED** [review]
Love the changes @Keddib . We remove one provider from the Didit SDK so the configuration is even easier for clients

File: packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx [hunk 2]
```diff
@@ -173,28 +174,12 @@ const DiditAuthProvider = ({
     if (token) {
       const token_info = parseJwt(token);
       if (token_info.exp * 1000 < Date.now()) {
-        removeToken();
-        removeAuthMethod();
+        deauthenticate();
```
[the remark this answers is not in the document]
Comment (pablogamium) [reply packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx]: I've merged the code and everything ir working fine for me, but please check it too @Keddib

File: packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx [hunk 3]
```diff
@@ -103,12 +103,12 @@ const DiditAuthProvider = ({
   const authenticate = useCallback(
     (_authMethod: DiditAuthMethod) => {
       setAuthMethod(_authMethod);
-
       if (status !== AuthenticationStatus.AUTHENTICATED) {
         setStatus(AuthenticationStatus.AUTHENTICATED);
       }
+      onLogin();
```
[the remark this answers is not in the document]
Comment (pablogamium) [reply packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx]: already solved

File: packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx [hunk 4]
```diff
@@ -173,28 +174,12 @@ const DiditAuthProvider = ({
...
-      status === AuthenticationStatus.AUTHENTICATED
-    ) {
-      onLogin(authMethod);
-    } else if (
-      prevStatus === AuthenticationStatus.AUTHENTICATED &&
-      status === AuthenticationStatus.UNAUTHENTICATED
-    ) {
-      onLogout();
-    }
-  }, [prevStatus, status, authMethod, onLogin, onLogout]);
```
[the remark this answers is not in the document]
Comment (pablogamium) [reply packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx:244]: Seems to we working fine now without it 👍

**Comment by rosasalberto** [comment]
Are these DiditAuthProvider props needed? What is the use of it? @pablogamium @Keddib 

```
emailAuthBaseUrl="https://my.didit.app/auth/email"
emailAuthorizationPath="/authorize/"
emailRedirectionPath="/redirect/"
```

**Review by pablogamium — APPROVED** [review]

File: packages/didit-sdk/src/hooks/useDiditAuth.ts (hunk 1 above)
Comment (Keddib) [reply packages/didit-sdk/src/hooks/useDiditAuth.ts:34]: @pablogamium ??
but we still want to use `useDiditAuth` hook if we are using only social auth method ?

File: packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Keddib) [reply packages/didit-sdk/src/contexts/diditAuthContext/DiditAuthProvider.tsx]: Works perfectly 🙏

File: README.md [hunk 5]
```diff
@@ -169,9 +132,46 @@ import { DiditAuthProvider, DiditAuthMethod } from 'didit-sdk';
       >
 ```
 
-2. Set up [`WagmiConfig`](https://wagmi.sh/docs/provider) provider and `DiditRainbowkitProvider`:
+##### Configure Wagmi
+
+if you want to use the wallet method, You will also need to configure your desired chains, generate the required connectors and setup a `wagmi` config
```
[the remark this answers is not in the document]
Comment (Keddib) [reply README.md]: fixed https://github.com/didit-protocol/didit-sdk/pull/44/commits/e2ff2f7fc58e6e935d192626fd13a1959d098b15

File: packages/didit-sdk/src/hooks/useDiditAuth.ts (hunk 1 above)
Comment (pablogamium) [reply packages/didit-sdk/src/hooks/useDiditAuth.ts:34]: Yes, for simplicity you can always use the same hook and just retrieve desired props

**Comment by pablogamium** [comment]
> Are these DiditAuthProvider props needed? What is the use of it? @pablogamium @Keddib
> 
> ```
> emailAuthBaseUrl="https://my.didit.app/auth/email"
> emailAuthorizationPath="/authorize/"
> emailRedirectionPath="/redirect/"
> ```

Ok, those are removed in the following PR: https://github.com/didit-protocol/didit-sdk/pull/45/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R122-R127

**Pull request merged**